web/src/main/java/org/cbioportal/web/GenesetController.java

```diff
@@ -70,5 +71,14 @@
         return new ResponseEntity<>(genesetService.getGeneset(genesetId), HttpStatus.OK);
     }
 
+    @RequestMapping(value = "/genesets/fetch", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE,
+	        produces = MediaType.APPLICATION_JSON_VALUE)
+	    @ApiOperation("Fetch gene sets by ID")
+	    public ResponseEntity<List<Geneset>> fetchGenesets(
+	        @ApiParam(required = true, value = "List of Gene set IDs")
+	        @RequestBody List<String> genesetIds) {
```

Comment:
Can you add a size constraint like https://github.com/cBioPortal/cbioportal/blob/master/web/src/main/java/org/cbioportal/web/PatientController.java#L90